### DISCUSSION TITLE
CYFS stack should support interfaces for NDN upload task

### DISCUSSION THREAD

**Discussion by photosssa** [buckyos/CYFS#78] [Ideas]
# CYFS stack has supported interfaces for NDN download task
After a download task created by get_data or create_trans_task, CYFS stack support following interfaces for it: 
+ Group tasks with path
+ Control state of task or group, start/pause/resume/cancel it
+ Query state of task or group, like progress, speed

# CYFS stack should support interfaces for NDN upload task
When a dec app based on CYFS stack got interest request for a chunk from a remote download task, if app decides to response chunk's data, a upload task should be created. Similar to interfaces for a download task, app may interact with upload tasks with interfaces:
+ Group tasks with path
+ Control state of task or group, start/pause/resume/cancel it
+ Query state of task or group, like progress, speed

## A scene: making a backup for a directory on device to its OOD
In this scene, service running on OOD downloads files from device, app running on device has to display backup progress, speed on user interface; 
We only supports interface for download task now, app will implement this scene by shortly polling download task's state from OOD with NON get request;
If interfaces for upload tasks supported, app can implement it simply by shortly polling upload task's state locally;

**Comment by tylittle92** [comment]
Hello everyone, I'm not sure if you can help with this or if this is even the right thread but I'm having issues with creating a user interface for a head unit that I'm building.  Can anyone help me out with this one?
